src/puya/ir/optimize/intrinsic_simplification.py

```diff
@@ -336,16 +382,31 @@ def _choose_encoding(a: AVMBytesEncoding, b: AVMBytesEncoding) -> AVMBytesEncodi
     return AVMBytesEncoding.base16
 
 
+def _decode_address(address: str) -> bytes:
+    """check if address is a valid address with checksum"""
```

Comment:
is this docstring incorrect? or should this function validate the length and hash?

Reply:
copy-pasta error